pkg/ottl/ottlfuncs/func_extract_grok_patterns.go

```diff
@@ -0,0 +1,94 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/go-grok"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ExtractGrokPatternsArguments[K any] struct {
+	Target             ottl.StringGetter[K]
+	Pattern            string
+	PatternDefinitions []string
```

Comment:
PatternDefinitions should still be an `Optional[[]string]` right?

Reply:
aah yes, got confused by when context propagation is needed